### PR TITLE
Reply integration

### PR DESCRIPTION
**Pull request by veken1199**
This pr includes UI reply implementation, database, and Handler changes

### CODE REVIEW

**Review by shelbaz — APPROVED** [review]
Perfect PR. Looks good.

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 1]
```diff
@@ -244,27 +245,32 @@
...
-  private long       threadId;
-  private int        distributionType;
-  private boolean    archived;
-  private boolean    isSecureText;
-  private boolean    isDefaultSms          = true;
-  private boolean    isMmsEnabled          = true;
-  private boolean    isSecurityInitialized = false;
-  private boolean    isSearchMode          = false;
-  private boolean    isEmojiReactionMode   = false;
+  private   Stub<EmojiDrawer>     emojiDrawerStub;
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationActivity.java:248]: Unaligned code

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 2]
```diff
@@ -609,6 +620,25 @@ private void handleReturnToConversationList() {
...
+    if(!isSearchMode) {
+      this.record = record;
+
+      bottomPanel = (InputPanel)findViewById(R.id.bottom_panel);
+      replyEdit = (EditText) findViewById(R.id.custom_reply);
+
+      replyEdit.setOnEditorActionListener(new ReplyInitiatedListener());
+
+      if (!this.isReplyMode) {
+        // We use the same input keyboard as search
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationActivity.java:633]: The comment is not needed

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 3]
```diff
@@ -836,21 +866,31 @@ public void handleNewReaction(MessageRecord messageRecord, ReactionsHandler hand
   }
 
   private void handleSearch(MenuItem item) {
-    bottomPanel = (InputPanel)findViewById(R.id.bottom_panel);
-    searchView = (SearchView)findViewById(R.id.custom_search);
+    if(!isReplyMode) {
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationActivity.java:869]: this is to make sure that users can't search while replying

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 4]
```diff
@@ -861,6 +901,13 @@ public void hideSearchMode() {
     fragment.resetAdapterSearchHandler();
   }
 
+  public void showReplyMode(InputPanel i, EditText s) {
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationActivity.java:904]: 's' parameter is never used, the method should be refactored

File: src/org/thoughtcrime/securesms/ConversationActivity.java [hunk 5]
```diff
@@ -2157,6 +2214,41 @@ public boolean onQueryTextChange(String newText) {
...
+          Map<String, String> replyBody = new HashMap<>();
+          replyBody.put("type", Stereotype.REPLY);
+          replyBody.put("hash", record.getHash());
+          replyBody.put("reply", v.getText().toString());
+          replyBody.put("time", time.toString());
+          replyBody.put("address", myAddress.serialize());
+
+          String body = new JSONObject(replyBody).toString();
+
+          // TODO check if it group chat reply
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationActivity.java:2239]: to be removed

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 6]
```diff
@@ -134,6 +140,7 @@
   private DeliveryStatusView deliveryStatusIndicator;
   private AlertView          alertView;
   private FlexboxLayout      reactionsList;
+  private LinearLayout       commentsList;
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationItem.java:143]: this is not used, extra member variable

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 7]
```diff
@@ -639,32 +647,100 @@ private String getMessageSenderName(Address senderAddress) {
...
-        for (Recipient participant : messageRecord.getIndividualRecipient().getParticipants()) {
-          if (participant.getAddress().serialize().equals(senderAddress.serialize())) {
-            messageSenderName = participant.getName() == null ? participant.getAddress().serialize() : participant.getName();
-            break;
-          }
-        }
-      } else {
-        if (messageSenderName == null) {
-          messageSenderName = messageRecord.getRecipient().getAddress().toString();
+        messageSenderName = messageRecord.getRecipient().getName();
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationItem.java:658]: alignment issue

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 8]
```diff
@@ -639,32 +647,100 @@ private String getMessageSenderName(Address senderAddress) {
...
+
+        TextView tvReceipient = theInflatedView.findViewById(R.id.pinned_message_recipient);
+        TextView tvMessage    = theInflatedView.findViewById(R.id.pinned_message_body);
+        TextView tvTime       = theInflatedView.findViewById(R.id.conversation_item_date);
+
+        tvMessage.setText(reply.getReply());
+        tvTime.setText(DateUtils.getExtendedRelativeTimeSpanString(context, new Locale("en", "CA"),
+                reply.getReplyDate()));
+
+        // check if it is me replying
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationItem.java:709]: This code is extra and can be replaced by `getMessageSenderName()`

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 9]
```diff
@@ -639,32 +647,100 @@ private String getMessageSenderName(Address senderAddress) {
...
+          tvReceipient.setText(messageSenderName);
+          this.setReplyViewOrientation(messageRecord,
+                  theInflatedView.findViewById(R.id.pinned_message_wrapper), ALIGN_PARENT_LEFT);
+        }
+        theInflatedView.setBackgroundColor(Color.parseColor("#E0E0E0"));
+        replyList.addView(theInflatedView);
+      }
+  }
+
+  private void setReplyViewOrientation(MessageRecord record, View theInflatedView, int alignParentRight) {
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationItem.java:738]: `record` is extra. should be removed

File: src/org/thoughtcrime/securesms/database/DatabaseFactory.java [hunk 10]
```diff
@@ -113,13 +113,14 @@
...
   private static final int UNSEEN_NUMBER_OFFER                             = 43;
   private static final int READ_RECEIPTS                                   = 44;
-  private static final int GROUP_RECEIPT_TRACKING                          = 45;
-  private static final int UNREAD_COUNT_VERSION                            = 46;
-  private static final int MORE_RECIPIENT_FIELDS                           = 47;
-  private static final int INTRODUCED_PINNED_MESSAGES                      = 48;
-  private static final int INTRODUCED_NICKNAMES                            = 49;
-  private static final int INTRODUCED_MESSAGE_REACTION                     = 50;
-  private static final int DATABASE_VERSION                                = 50;
+  private static final int GROUP_RECEIPT_TRACKING      = 45;
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/database/DatabaseFactory.java:116]: Alignment issue

**Review by veken1199 — COMMENTED** [review]
The changes will pushed to a different branch that will target code quality

**Pull request merged**